server/conn.go

```diff
@@ -693,6 +744,15 @@ func (cc *clientConn) openSessionAndDoAuth(authData []byte) error {
 	if err = cc.server.checkConnectionCount(); err != nil {
 		return err
 	}
+	return nil
+}
+
+func (cc *clientConn) openSessionAndDoAuth(authData []byte) error {
```

Comment:
If open session success, but do auth fails, will there be a potential resource leak?